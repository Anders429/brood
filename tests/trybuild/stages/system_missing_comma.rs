use brood::{query::{filter, result, views}, registry::ContainsQuery, system::{schedule::stages, System}};

struct MySystem;

impl System for MySystem {
    type Views<'a> = views!();
    type Filter = filter::None;

    fn run<'a, R, FI, VI, P, I, Q>(&mut self, query_results: result::Iter<'a, R, Self::Filter, FI, Self::Views<'a>, VI, P, I, Q>)
    where
        R: ContainsQuery<'a, Self::Filter, FI, Self::Views<'a>, VI, P, I, Q>,
    {}
}

type Stages = stages!{
    system: MySystem
    flush
};

fn main() {}
