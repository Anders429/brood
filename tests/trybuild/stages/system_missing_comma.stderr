error: no rules expected the token `flush`
  --> tests/trybuild/stages/system_missing_comma.rs:18:5
   |
18 |     flush
   |     ^^^^^ no rules expected this token in macro call
